src/main/java/frc/utils/math/MirrorMath.java

```diff
@@ -18,4 +21,37 @@ public static double getMirroredY(double y) {
 		return Field.WIDTH_METERS - y;
 	}
 
+	public static Pose2d getMirrored(Pose2d pose2d, boolean mirrorY, boolean mirrorX, boolean invertAngle) {
+		if (mirrorY && mirrorX) {
+			return invertAngle
+				? new Pose2d(getMirroredX(pose2d.getX()), getMirroredY(pose2d.getY()), getMirroredAngle(pose2d.getRotation()))
+				: new Pose2d(getMirroredX(pose2d.getX()), getMirroredY(pose2d.getY()), pose2d.getRotation());
+		}
+		if (mirrorY) {
+			return invertAngle
+				? new Pose2d(pose2d.getX(), getMirroredY(pose2d.getY()), getMirroredAngle(pose2d.getRotation()))
+				: new Pose2d(pose2d.getX(), getMirroredY(pose2d.getY()), pose2d.getRotation());
+		}
+		if (mirrorX) {
+			return invertAngle
+				? new Pose2d(getMirroredX(pose2d.getX()), pose2d.getY(), getMirroredAngle(pose2d.getRotation()))
+				: new Pose2d(getMirroredX(pose2d.getX()), pose2d.getY(), pose2d.getRotation());
+		}
+		return invertAngle
+			? new Pose2d(pose2d.getX(), pose2d.getY(), getMirroredAngle(pose2d.getRotation()))
+			: new Pose2d(pose2d.getX(), pose2d.getY(), pose2d.getRotation());
+	}
```

Comment:
```suggestion
		Pose2d output = pose2d;
		if (mirrorY) {
			output = new Pose2d(getMirroredX(pose2d.getX()), pose2d.getY(), pose2d.getrotation();
		}
		if (mirrorX) {
			output = new Pose2d(pose2d.getX(), getMirroredY(pose2d.getY()), pose2d.getrotation();
		}
		if (inverAngle) {
			output = new Pose2d(output.getTranslation(), output.getRotation().unaryMinus)
		}
		return output;
```

requires $n$ if statements instead of $n!$